src/System.ComponentModel.TypeConverter/tests/BindingListTests.cs

```diff
@@ -0,0 +1,699 @@
+				list_changed = true;
+				change_type = e.ListChangedType;
+				list_changed_index = e.NewIndex;
+			};
+
+			object rv = l.AddNew ();
+			Assert.True (adding_event_raised, "2");
+			Assert.NotNull (rv);
+
+			Assert.Equal (1, l.Count);
+			Assert.Equal (0, l.IndexOf (rv));
+			Assert.True (list_changed, "6");
+			Assert.Equal (ListChangedType.ItemAdded, change_type);
+			Assert.Equal (0, list_changed_index);
+
+			list_changed = false;
+
+			l.EndNew (2);
+
+			Assert.Equal (1, l.Count);
+			Assert.False (list_changed, "10");
+
+			l.CancelNew (0);
+
+			Assert.True (list_changed, "11");
+			Assert.Equal (ListChangedType.ItemDeleted, change_type);
+			Assert.Equal (0, list_changed_index);
+		}
+
+		class BindingListPoker : BindingList<object>
+		{
+			public object DoAddNewCore()
+			{
+				return base.AddNewCore ();
+			}
+		}
+
+		// test to make sure that the events are raised in AddNewCore and not in AddNew
+		[Fact]
+		public void TestAddNewCore_Insert ()
+		{
+			BindingListPoker poker = new BindingListPoker ();
+
+			bool adding_event_raised = false;
+
+			bool list_changed = false;
+			ListChangedType change_type = ListChangedType.Reset;
+			int list_changed_index = -1;
+
+			poker.AddingNew += delegate (object sender, AddingNewEventArgs e) {
+				adding_event_raised = true;
+			};
+
+			poker.ListChanged += delegate (object sender, ListChangedEventArgs e) {
+				list_changed = true;
+				change_type = e.ListChangedType;
+				list_changed_index = e.NewIndex;
+			};
+
+			object o = poker.DoAddNewCore ();
+
+			Assert.True (adding_event_raised, "1");
+			Assert.True (list_changed, "2");
+			Assert.Equal (ListChangedType.ItemAdded, change_type);
+			Assert.Equal (0, list_changed_index);
+			Assert.Equal (1, poker.Count);
+		}
+
+		private class Item : INotifyPropertyChanged {
+
+			public event PropertyChangedEventHandler PropertyChanged;
+
+			string _name;
+
+			public string Name {
+				get { return _name; }
+				set {
+					if (_name != value) {
+						_name = value;
+						OnPropertyChanged ("Name");
+					}
+				}
+			}
+
+			void OnPropertyChanged (string name)
+			{
+				var fn = PropertyChanged;
+				if (fn != null)
+					fn (this, new PropertyChangedEventArgs (name));
+			}
+		}
+
+		// [Fact] // https://bugzilla.xamarin.com/show_bug.cgi?id=8366
+		// public void Bug8366 ()
+		// {
+		// 	bool added = false;
+		// 	bool changed = false;
+		// 	var list = new BindingList<Item> ();
+		// 	list.ListChanged += (object sender, ListChangedEventArgs e) => {
+		// 		added |= e.ListChangedType == ListChangedType.ItemAdded;
+		// 		changed |= e.ListChangedType == ListChangedType.ItemChanged;
+		// 	};
+
+		// 	var item = new Item () { Name = "1" };
+		// 	list.Add (item);
+
+		// 	item.Name = "2";
+
+		// 	Assert.True (added, "ItemAdded");
+		// 	Assert.True (changed, "ItemChanged");
+		// }
+		
+		[Fact] // https://bugzilla.xamarin.com/show_bug.cgi?id=16902
+		public void Bug16902 ()
```

Comment:
These seem to be tests for bugs in a different implementation.  The test may still be a good thing to have, but it'd be good to choose a name that reflects what it's actually testing, rather than a bug number for what's effectively a separate project.